app/client/components/delivery/records/deliveryRecords.tsx

```diff
@@ -260,6 +260,28 @@ export const DeliveryRecordsFC = (props: DeliveryRecordsFCProps) => {
     ...(hasRecentHolidayStop ? [holidaySuspensionDeliveryProblem] : [])
   ].sort((a, b) => a.label.localeCompare(b.label));
 
+  const formErrorTitle =
+    !step3FormValidationDetails.isValid &&
+    step1FormValidationDetails.isValid &&
+    step2FormValidationDetails.isValid
+      ? "Unfinished changes"
+      : "Some information is missing";
+
+  const formErrorMessages = [
+    ...(!step1FormValidationDetails.isValid &&
+    step1FormValidationDetails.message
+      ? [step1FormValidationDetails.message]
+      : []),
+    ...(!step2FormValidationDetails.isValid &&
+    step2FormValidationDetails.message
+      ? [step2FormValidationDetails.message]
+      : []),
+    ...(!step3FormValidationDetails.isValid &&
+    step3FormValidationDetails.message
+      ? [step3FormValidationDetails.message]
+      : [])
+  ];
```

Comment:
Would using `formErrorMessages.push()` help simplify this code?

Reply:
As the array contains 3 elements with repeated logic this method seems feasible. The alternative of a chain of if statements with `formErrorMessages.push()` would make the code more verbose. Though, this method does require JS knowledge on how the spread operator works on an array within an array.

Reply:
this looks ripe for a function which takes in `validationDetails` does the ternary fun, which you could either have as a named function, which you call three times, or better IMO would be something like...
```suggestion
  const formErrorMessages = [
       step1FormValidationDetails,
       step2FormValidationDetails,
       step3FormValidationDetails
    ].filter(validationDetails => 
        !validationDetails.isValid && validationDetails.message
    );
```

Reply:
Ooh that looks great! 👍  
As I'm new here, may I commit your suggestion here? Or do I simply praise it whilst making another commit locally and pushing it to this branch

Reply:
Hi Maria, and welcome!

Good question, in my experience for small typo style suggestions, committing directly from GitHub suggestion (using the `Commit suggestion`). However for something like this I would probably reimplement the change yourself to test it works as expected and to ensure prettier does its thing etc.

Worth saying though, its totally up to you there is no rule (or even convention) and this is your PR (that you could rebase if you committed a suggestion you later wished you hadn't.

Only thing worth nothing, is that committing suggestions adds the suggester as a[ 'Co-Author'](https://docs.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors) which you might want to bear in mind when using this in future (I'm not mentioning because I want credit for this suggestion, for my sins my name is all over this repo already 🙈 )

Reply:
FormErrorMessages has been refactored